drivers/ipm/ipm_mhu.c

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <device.h>
+#include <soc.h>
+#include "ipm_mhu.h"
+
+#define DEV_CFG(dev) \
+	((const struct ipm_mhu_device_config * const)(dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct ipm_mhu_data *)(dev)->driver_data)
+#define IPM_MHU_REGS(dev) \
+	((volatile struct ipm_mhu_reg_map_t *)(DEV_CFG(dev))->base)
+
+#if defined(CONFIG_IPM_MHU0) || defined(CONFIG_IPM_MHU1)
+
+static enum ipm_mhu_cpu_id_t ipm_mhu_get_cpu_id(const struct device *d)
+{
+	volatile u32_t *p_mhu_dev_base;
+	volatile u32_t *p_cpu_id;
+
+	p_mhu_dev_base = (volatile u32_t *)IPM_MHU_REGS(d);
+
+	p_cpu_id = (volatile u32_t *)(((u32_t)p_mhu_dev_base &
+						SSE_200_DEVICE_BASE_REG_MSK) +
+						SSE_200_CPU_ID_UNIT_OFFSET);
+
+	return (enum ipm_mhu_cpu_id_t)*p_cpu_id;
+}
+
+static u32_t ipm_mhu_get_status(const struct device *d,
+						enum ipm_mhu_cpu_id_t cpu_id,
+						u32_t *status)
+{
+	struct ipm_mhu_reg_map_t *p_mhu_dev;
+
+	if (status == NULL) {
+		return IPM_MHU_ERR_INVALID_ARG;
+	}
+
+	p_mhu_dev = (struct ipm_mhu_reg_map_t *)IPM_MHU_REGS(d);
+
+	switch (cpu_id) {
+	case IPM_MHU_CPU1:
+		*status = p_mhu_dev->cpu1intr_stat;
+		break;
+	case IPM_MHU_CPU0:
+	default:
+		*status = p_mhu_dev->cpu0intr_stat;
+		break;
+	}
+
+	return IPM_MHU_ERR_NONE;
+}
+
+static int ipm_mhu_send(struct device *d, int wait, u32_t cpu_id,
+			  const void *data, int size)
+{
+	ARG_UNUSED(wait);
+	u32_t set_val = *(u32_t *)data;
```

Comment:
The register value should not from the param.
Suggest to use a fixed number instead:

	ARG_UNUSED(data);
	u32_t set_val = 0x01;